### DISCUSSION TITLE
DLL load failed error when executing encrypted Pyarmor code (windows)

### DISCUSSION THREAD

**Discussion by nrtoribio** [dashingsoft/pyarmor#1345] [Q&A] [answered]
I am encountering an issue while attempting to execute encrypted code using Pyarmor on a Windows machine. Here are the details of the problem:

- License: We have a Pyarmor Group license.
- Encryption Process: On a Windows 11 with Python 3.10 and pyarmor 8.2.4, we encrypted our code using the command "pyarmor gen hello_world.py". This process successfully generated the dist folder, containing the encrypted code and the pyarmor_runtime_006024 folder.
- Transfer to another machine: We copied the dist folder generated by Pyarmor onto another Windows machine for execution.
- Error Message: However, when we try to execute the code using the command "python hello_world.py", we encounter the following error:

python hello_world.py
Traceback (most recent call last):
  File "C:\Users\test\hello_world_pyarmor (1)\hello_world.py", line 2, in <module>
    from pyarmor_runtime_006024 import __pyarmor__
  File "C:\Users\test\hello_world_pyarmor (1)\pyarmor_runtime_006024\__init__.py", line 2, in <module>
    from .pyarmor_runtime import __pyarmor__
ImportError: DLL load failed while importing pyarmor_runtime: The specified module could not be found.

 
We are seeking assistance in understanding the cause of this error and any potential solutions. I would appreciate any guidance or insights that can help resolve this issue. Thank you!

**Comment by jondy** [comment]
Generally it's different Python version issue. The target machine is not Python 3.10.

Also upgrade Pyarmor to latest version

Please check documentation to understand more about pyarmor.

**Reply by nrtoribio** [reply]
Hello again. Thank you, the target machine was not Python 3.10 but 3.9. 
Now we can run hello_world.py with success.
However, when trying to run our code, we encounter the error:
![image](https://github.com/dashingsoft/pyarmor/assets/137907517/d9a06aa1-cf59-4231-b338-1a78fe7302a2)

The difference with hello_world.py is that we encrypted this code with an expiration date using "pyarmor gen -e 2023-12-31 Code". I can successfully run the encrypted code on my local computer but not on the target machine. Plus, the target machine does not have internet access, is it required to run an encrypted code with an expiration date?

**Comment by jondy** [comment, marked as answer]
I suggest first have a glance on [documentation](https://pyarmor.readthedocs.io/en/stable/) to understand Pyarmor features. In this case, just check the description of option `--expired`.
